### PR TITLE
feat: support 7702 transactions

### PR DESCRIPTION
**Pull request by antoniolocascio**
# What ❔

## Checklist

- [ ] PR title corresponds to the body of PR.
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `cargo fmt` and checked with `cargo clippy`.

### CODE REVIEW

File: indexes/zk_ee/ethereum-execution-specs-general-state-tests.yaml [hunk 1]
```diff
@@ -647,7 +648,8 @@ entries:
                 enabled: true
               contract_create.json:
                 hash: '0xc60f5f9fd386b9d978b02a41f10ff0df'
-                enabled: true
+                enabled: false
+                comment: "Framework doesn't let us create a 7702 tx with null to"
```
Comment (0xVolosnikov) [inline indexes/zk_ee/ethereum-execution-specs-general-state-tests.yaml:652]: Maybe we should add "TODO" here?

**Review by 0xVolosnikov — APPROVED** [review]

File: indexes/zk_ee/ethereum-execution-specs-general-state-tests.yaml (hunk 1 above)
Comment (antoniolocascio) [reply indexes/zk_ee/ethereum-execution-specs-general-state-tests.yaml:652]: Not sure, the alloy TransactionRequest breaks when trying to build a 7702 tx with null to. I was happy to rely on alloy TransactionRequest to interface tester and api. Don't think it's worth doing more work just for this test.

Comment (antoniolocascio) [reply indexes/zk_ee/ethereum-execution-specs-general-state-tests.yaml:652]: To be extra safe, added a test for to=null to instances/transactions
